File: app/src/main/java/com/mediaplayer/services/MobileArrayAdapter.java
package com.mediaplayer.services;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.mediaplayer.R;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.util.List;

public class MobileArrayAdapter extends ArrayAdapter<String> {
    private final Context context;
    private final List<String> videos;
    private final List<String> thumb;
    private final List<Long> time;
    private final List<Long> filesize;

    public MobileArrayAdapter(Context context, List<String> videos, List<String> thumb, List<Long> time, List<Long> filesize) {
        super(context, R.layout.thumb_list, videos);
        this.context = context;
        this.videos = videos;
        this.thumb = thumb;
        this.time = time;
        this.filesize = filesize;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View rowView = inflater.inflate(R.layout.thumb_list, parent, false);
        TextView textView = (TextView) rowView.findViewById(R.id.th_text_view);
        ImageView imageView = (ImageView) rowView.findViewById(R.id.icon);
        TextView time_text = (TextView) rowView.findViewById(R.id.time_text);
        TextView text_file_size = (TextView) rowView.findViewById(R.id.text_file_size);
        textView.setText(new File(videos.get(position)).getName());
        time_text.setText(new MathService().timeFormatter(time.get(position)));
        text_file_size.setText(new MathService().convertFileSize(filesize.get(position)));
        Bitmap bitmap = null;
        File f = new File(thumb.get(position));
        BitmapFactory.Options options = new BitmapFactory.Options();
        options.inPreferredConfig = Bitmap.Config.ARGB_8888;
        try {
            bitmap = BitmapFactory.decodeStream(new FileInputStream(f), null, options);
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
        imageView.setImageBitmap(bitmap);
        return rowView;
    }
}

File: app/src/main/java/com/mediaplayer/services/MobileArrayAdapter.kt
package com.mediaplayer.services

import android.content.Context
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView

import com.mediaplayer.R

import java.io.File
import java.io.FileInputStream
import java.io.FileNotFoundException

class MobileArrayAdapter(context: Context, private val videos: List<String>, private val thumb: List<String>, private val time: List<Long>, private val filesize: List<Long>) : ArrayAdapter<String>(context, R.layout.thumb_list, videos) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val rowView = inflater.inflate(R.layout.thumb_list, parent, false)
        val textView = rowView.findViewById<View>(R.id.th_text_view) as TextView
        val imageView = rowView.findViewById<View>(R.id.icon) as ImageView
        val time_text = rowView.findViewById<View>(R.id.time_text) as TextView
        val text_file_size = rowView.findViewById<View>(R.id.text_file_size) as TextView
        textView.text = File(videos[position]).name
        time_text.text = MathService.timeFormatter(time[position])
        text_file_size.text = MathService.convertFileSize(filesize[position])
        loadImageInThread(position, imageView)
        return rowView
    }

    fun loadImageInThread(position: Int, imageView: ImageView) {
        var bitmap: Bitmap?
        if (thumb[position] != null) {
            val f = File(thumb[position])
            val options = BitmapFactory.Options()
            options.inPreferredConfig = Bitmap.Config.ARGB_8888
            try {
                bitmap = BitmapFactory.decodeStream(FileInputStream(f), null, options)
            } catch (e: FileNotFoundException) {
                bitmap = BitmapFactory.decodeResource(context.resources, R.drawable.play_circle_outline)
                println("File not found exception:::: $e")
            }

        } else {
            bitmap = BitmapFactory.decodeResource(context.resources, R.drawable.play_circle_outline)
        }
        imageView.setImageBitmap(bitmap)
    }
}